src/Tribe/Date_Utils.php

```diff
@@ -49,6 +48,109 @@ public static function datepicker_formats( $translate = null ) {
 		}
 
 		/**
+		 * As PHP 5.2 doesn't have a good version of `date_parse_from_format`, this is how we deal with
+		 * possible weird datepicker formats not working
+		 *
+		 * @param  string $format The weird format you are using
+		 * @param  string $date   The date string to parse
+		 *
+		 * @return string         A DB formated Date, includes time if possible
+		 */
+		public static function time_from_format( $format, $date ) {
+			// Reverse engineer the relevant date formats
+			$keys = array(
+				// Year with 4 Digits
+				'Y' => array( 'year', '\d{4}' ),
+
+				// Year with 2 Digits
+				'y' => array( 'year', '\d{2}' ),
+
+				// Month with leading 0
+				'm' => array( 'month', '\d{2}' ),
+
+				// Month without the leading 0
+				'n' => array( 'month', '\d{1,2}' ),
+
+				// Month ABBR 3 letters
+				'M' => array( 'month', '[A-Z][a-z]{3}' ),
+
+				// Month Name
+				'F' => array( 'month', '[A-Z][a-z]{2,8}' ),
+
+				// Day with leading 0
+				'd' => array( 'day', '\d{2}' ),
+
+				// Day without leading 0
+				'j' => array( 'day', '\d{1,2}' ),
+
+				// Day ABBR 3 Letters
+				'D' => array( 'day', '[A-Z][a-z]{2}' ),
+
+				// Day Name
+				'l' => array( 'day', '[A-Z][a-z]{6,9}' ),
+
+				// Hour 12h formatted, with leading 0
+				'h' => array( 'hour', '\d{2}' ),
+
+				// Hour 24h formatted, with leading 0
+				'H' => array( 'hour', '\d{2}' ),
+
+				// Hour 12h formatted, without leading 0
+				'g' => array( 'hour', '\d{1,2}' ),
+
+				// Hour 24h formatted, without leading 0
+				'G' => array( 'hour', '\d{1,2}' ),
+
+				// Minutes with leading 0
+				'i' => array( 'minute', '\d{2}' ),
+
+				// Seconds with leading 0
+				's' => array( 'second', '\d{2}' ),
+			);
+
+			// Convert format string to regex
+			$regex = '';
+			$chars = str_split( $format );
+			foreach ( $chars as $n => $char ) {
+				$last_char = isset( $chars[ $n - 1 ] ) ? $chars[ $n - 1 ] : '';
+				$skip_current = '\\' == $last_char;
+				if ( ! $skip_current && isset( $keys[ $char ] ) ) {
+					$regex .= '(?P<' . $keys[ $char ][0] . '>' . $keys[ $char ][1] . ')';
```

Comment:
Here we are looking for the `$char` key within the `$keys` that is defined on the beginning of the function, this prevents non supported date Chars to be converted to a _regex_.